### PR TITLE
Fix permission check for net

### LINKED ISSUE #2371: [Question] Is the incompatibility of allow-net between flags and options intentionally?
## Overview

Is the incompatibility of allow-net between flags and options intentionally?

## Details
I found the incompatiblity behavior of `--allow-net`.

When using `--allow-net` as flag, we need no additional permission.

```
deno run --allow-net https://deno.land/std/examples/echo_server.ts
```

The other hand, using `--allow-net` as options, we need additional permission about "listen" and "accept".

```
deno run --allow-net=0.0.0.0 https://deno.land/std/examples/echo_server.ts
⚠️  Deno requests network access to "listen". Grant? [a/y/n/d (a = allow always, y = allow once, n = deny once, d = deny always)]
⚠️  Deno requests network access to "accept". Grant? [a/y/n/d (a = allow always, y = allow once, n = deny once, d = deny always)]
```

Is the behavior is intentionally?
Does whitelist permission of `--allow-net` affect only `fetch`?

## environment
deno: 0.5.0
v8: 7.6.53
typescript: 3.4.1

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #2371

It turns out we were checking for "listen", "dial" and "accept" addresses in net permissions 😅 

I removed check in `op_accept` because we need to first issue `op_listen` and there is already check there.

### CODE REVIEW

**Comment by ry** [comment]
Thanks! Is it possible to add a test to tools/complex_permissions_test.py that demos this fix? (Presumably, for example, listen wasn't properly obeying --allow-net arguments before.)

**Comment by bartlomieju** [comment]
Sorry @ry, should have started with tests in the first place, turns out my initial solution was wrong 😉 I guess it's worth mentioning in the docs that `--allow-net=addr1,addr2` allows to access `addr1` and `addr2` before they are resolved

File: tools/complex_permissions_test.py [hunk 1]
```diff
@@ -80,31 +81,68 @@ def warm_up(self):
...
+        # wrap_test(test_name_base + "_inside_tests_dir",
+        #           self.test_inside_test_dir, test_type)
+        # wrap_test(test_name_base + "_outside_tests_and_js_dir",
+        #           self.test_outside_test_and_js_dir, test_type)
+        # wrap_test(test_name_base + "_inside_tests_and_js_dir",
+        #           self.test_inside_test_and_js_dir, test_type)
+        # wrap_test(test_name_base + "_relative", self.test_relative,
+        #           test_type)
+        # wrap_test(test_name_base + "_no_prefix", self.test_no_prefix,
+        #           test_type)
```
Comment (ry) [inline tools/complex_permissions_test.py]: What's happening here?

Comment (bartlomieju) [reply tools/complex_permissions_test.py]: Yikes, sorry!

**Pull request merged**